website/src/components/Submission/FileUpload/FolderUploadComponent.tsx

```diff
@@ -0,0 +1,447 @@
+                                            type: 'pending',
+                                            file: file.file,
+                                            name: file.name,
+                                            size: file.file.size,
+                                            url: val[i].url,
+                                            fileId: val[i].fileId,
+                                        });
+                                        i++;
+                                    });
+                                });
+                                setFileUploadState({
+                                    type: 'uploadInProgress',
+                                    files: result,
+                                });
+
+                                Object.entries(result).forEach(([submissionId, files]) => {
+                                    files.forEach(({ file, url, fileId }) => {
+                                        fetch(url, {
+                                            method: 'PUT',
+                                            headers: {
+                                                // eslint-disable-next-line @typescript-eslint/naming-convention
+                                                'Content-Type': file.type,
+                                            },
+                                            body: file,
+                                        })
+                                            .then((response) => {
+                                                setFileUploadState((state) => {
+                                                    if (state?.type === 'uploadInProgress') {
+                                                        return produce(state, (draft) => {
+                                                            draft.files[submissionId] = state.files[submissionId].map(
+                                                                (file) => {
+                                                                    if (
+                                                                        file.type === 'pending' &&
+                                                                        file.fileId === fileId
+                                                                    ) {
+                                                                        if (response.ok) {
+                                                                            return {
+                                                                                type: 'uploaded',
+                                                                                fileId: file.fileId,
+                                                                                name: file.name,
+                                                                                size: file.size,
+                                                                            };
```

Comment:
Maybe break this into functions or multiple files, it's hard to review in Github UI as long lines get broken
<img width="766" alt="image" src="https://github.com/user-attachments/assets/09b3c6b4-cc5a-4662-9d2b-a6fab45fa418" />

Reply:
Oh yes, definitely. i've split out a function for "start uploading"